lib/handler/compress/gzip.c

```diff
@@ -74,16 +81,16 @@ static size_t compress_chunk(struct st_gzip_context_t *self, const void *src, si
         }
         self->zs.next_out = (void *)(self->bufs.entries[bufindex].base + self->bufs.entries[bufindex].len);
         self->zs.avail_out = (unsigned)(BUF_SIZE - self->bufs.entries[bufindex].len);
-        ret = deflate(&self->zs, flush);
-        assert(ret == Z_OK || ret == Z_STREAM_END);
+        ret = proc(&self->zs, flush);
+        assert(ret == Z_OK || ret == Z_STREAM_END || ret == Z_BUF_ERROR);
```

Comment:
I think we had the discussion before, but could you please once more explain why we need to take care of `Z_BUF_ERROR`?

The intention of the original code was to always supply enough space to zlib so that it would never return `Z_BUF_ERROR`. I would like to understand why you needed to accept `Z_BUF_ERROR`, without growing the amount of free space supplied to zlib.

Note that current design secures at least 32 bytes of room in the out buffer (see the `if` block above). If there is more than that, then current code does not try to make the room larger. In other words, our design does not follow the requirement of `deflate` or `inflate`, that states, quote: _if deflate returns with Z_OK or Z_BUF_ERROR, this function must be called again with Z_FINISH and more output space (updated avail_out) but no more input data, until it returns with Z_STREAM_END or an error_ (source: http://www.zlib.net/manual.html).